packages/nextjs/src/utils/withSentry.ts

```diff
@@ -79,53 +79,12 @@ export const withSentry = (origHandler: NextApiHandler): WrappedNextApiHandler =
-        //
-        // - In order to trick nextjs into not complaining, we can set `res.finished` to `true` before exiting the
-        //   handler. If we do that, though, `res.end()` gets mad because it thinks *it* should be the one to get to
-        //   mark the response complete. We therefore need to flip it back to `false` 1) after nextjs's check but 2)
-        //   before the original `res.end()` is called.
-        //
-        // - The second part is easy - we control when the original `res.end()` is called, so we can do the flipping
-        //   right beforehand and `res.end()` will be none the wiser.
-        //
-        // - The first part isn't as obvious. How do we know we won't end up with a race condition, such that the
-        //   flipping to `false` might happen before the check, negating the entire purpose of this hack? Fortunately,
-        //   before it's done, our async `res.end()` wrapper has to await a `setImmediate()` callback, guaranteeing its
-        //   run lasts at least until the next event loop. The check, on the other hand, happens synchronously,
-        //   immediately after the request handler (so in the same event loop). So as long as we wait to flip
-        //   `res.finished` back to `false` until after the `setImmediate` callback has run, we know we'll be safely in
-        //   the next event loop when we do so.
-        //
-        // And with that, everybody's happy: Nextjs doesn't complain about an unfinished response, `res.end()` doesn’t
-        // complain about an already-finished response, and we have time to make sure events are flushed to Sentry.
-        //
-        // One final note: It might seem like making `res.end()` an awaited async function would run the danger of
-        // having the lambda close before it's done its thing, meaning we *still* might not get events sent to Sentry.
-        // Fortunately, even though it's called `res.end()`, and even though it's normally sync, a) it's far from the
-        // end of the request process, so there's other stuff which needs to happen before the lambda can close in any
-        // case, and b) that other stuff isn't triggered until `res.end()` emits a `prefinished` event, so even though
-        // it's not technically awaited, it's still the case that the process can't go on until it's done.
-        //
-        // See
-        // https://github.com/vercel/next.js/blob/e1464ae5a5061ae83ad015018d4afe41f91978b6/packages/next/server/api-utils.ts#L106-L118
-        // and
-        // https://github.com/nodejs/node/blob/d8f1823d5fca5e3c00b19530fb15343fdd3c8bf5/lib/_http_outgoing.js#L833-L911.
-        res.finished = true;
+        if (process.env.NODE_ENV === 'development') {
+          // eslint-disable-next-line no-console
+          console.warn(
+            '\n[sentry] If Next.js logs a warning "API resolved without sending a response", it\'s a false positive, which we\'re working to rectify.\n',
```

Comment:
We can’t use the logger here?

Also why are we emitting a new line at the start?

Reply:
The logger only shows up if the user has `debug: true` in their config, whereas the warning our message references shows up regardless.

As for the newline, in my testing the message was just kind of getting lost in all the logs, so I stuck some blank lines around it. Do you think it's an issue?

Reply:
I would prefer to remove the new lines to reduce the noise caused by sentry.

That’s primarily my worry here with the `console.warn`, that it will cause a lot of noise in the development environment (that can’t be turned off). With debug logging, it’s opt in from the user.

Reply:
I can take the newlines out, but bottom line, this creates exactly as much noise as the warning we're causing. It's just that with this logging, it's now

```
[sentry] There's about to be a warning - it's fine.
Warning - things might be broken.
```

rather than 

```
Warning - things might be broken.
```

As a user, I'd rather have that be effectively a two-line warning, including reassurance that I don't need to stress, rather than a one-line warning whose origin and importance I have to wonder about (unless I happen to have turned on debugging in Sentry, which the warning itself gives me no hint that I'd need to do).

The least noise, of course, would be if we weren't causing the warning in the first place, and that's what the original PR accomplished. But given that for the moment we're choosing annoying warning over potential breakage, I really do think it's better to at least let people know why it's there and that it's safe to ignore. Don't you?

Reply:
Yeah that’s good perspective

Reply:
Hello @AbhiPrasad @lobsterkatie, how are you? Is the potential fix being tracked in an issue? Combined with next/auth and swr for fetching the user profile it's triggering a massive amount of logs.
 
<img width="1151" alt="image" src="https://user-images.githubusercontent.com/10283686/157413964-b374c483-8685-4644-89bf-6224d0d53199.png">

Thank you for your hard work 👍

Reply:
Hey, @Fedeorlandau thanks for reaching out. Could you please open an issue with your setup details? (SDK version, next version, next config), it'll help a ton for us to debug.